### PR TITLE
Drools 1180 Migrate QE tests upstream.

### PR DESCRIPTION
**Pull request by baldimir**
This is third in a series of PRs that migrate some tests from QE regression test suite upstream (there will be more).

Migrated test classes that inherit from KieBasesTest class in QE regression test suite.

### CODE REVIEW

**Comment by winklerm** [comment]
Looks good to me, +1.

File: drools-test-coverage/test-suite/src/test/resources/org/drools/testcoverage/regression/fusionAfterBeforeTest.drl [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+end
+
+rule "BeforeMessageEvent"
+when
+	$message1 : MessageEvent() from entry-point "EventStream"
+	$message2 : MessageEvent(this before[1s, 10s] $message1) from entry-point "EventStream"
+then
+	$message1.getType(); // just to test that both messages are not null 
+	$message2.getType();
+end
```
Comment (psiroky) [inline drools-test-coverage/test-suite/src/test/resources/org/drools/testcoverage/regression/fusionAfterBeforeTest.drl:25]: Nitpicking here -- some of the DRL files use spaces and some tabs. Would be great to unify that (and not use tabs :))

File: drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/PropertyListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        printAndAssertListenersFromPeople(people);
+    }
+
+    private void printAndAssertListenersFromPeople(final Collection<Person> people) {
+        for (Person person : people) {
+            for (PropertyChangeListener listener : person.getBoundSupport().getPropertyChangeListeners()) {
+                LOGGER.info("Listener attached of type: " + listener);
+            }
+            // there should be no listeners
+            Assertions.assertThat(person.getBoundSupport().getPropertyChangeListeners().length).isEqualTo(0);
```
Comment (psiroky) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/PropertyListenerTest.java]: I think we could use something like `assertThat(list/array).isEmpty()`. Makes the assert a little bit more readable.

File: drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/MultipleKieBaseListenersTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        kieBase.addEventListener(listener);
+        kieBase.addEventListener(listener);
+        kieBase.addEventListener(listener);
+
+        Assertions.assertThat(kieBase.getKieBaseEventListeners().size()).isEqualTo(1);
+
+        kieBase.removeEventListener(listener);
+
+        Assertions.assertThat(kieBase.getKieBaseEventListeners().size()).isEqualTo(0);
```
Comment (psiroky) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/MultipleKieBaseListenersTest.java]: `assertThat(list/array).isEmpty()` should be a little bit more readable.

File: drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/InaccurateComparisonTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                "     m : Message( message < \"90201304122000000000000017\" )\n" +
+                " then \n" +
+                " end";
+
+        final KieBase kieBase = KieBaseUtil.getKieBaseAndBuildInstallModuleFromDrl(TestConstants.PACKAGE_REGRESSION,
+                kieBaseTestConfiguration, rule);
+        KieSession ksession = kieBase.newKieSession();
+
+        ksession.insert(new Message("90201304122000000000000015"));
+        Assertions.assertThat(1).isEqualTo(ksession.fireAllRules());
```
Comment (psiroky) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/InaccurateComparisonTest.java]: The actual/expected values should be swapped here I think.

**Comment by psiroky** [comment]
Looks good, just added few minor comments.

File: drools-test-coverage/test-suite/src/test/resources/org/drools/testcoverage/regression/fusionAfterBeforeTest.drl (hunk 1 above)
Comment (baldimir) [inline drools-test-coverage/test-suite/src/test/resources/org/drools/testcoverage/regression/fusionAfterBeforeTest.drl:25]: Fixed.

File: drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/PropertyListenerTest.java (hunk 2 above)
Comment (baldimir) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/PropertyListenerTest.java]: Fixed.

File: drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/MultipleKieBaseListenersTest.java (hunk 3 above)
Comment (baldimir) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/MultipleKieBaseListenersTest.java]: Fixed.

File: drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/InaccurateComparisonTest.java (hunk 4 above)
Comment (baldimir) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/regression/InaccurateComparisonTest.java]: Fixed.

**Comment by psiroky** [comment]
+1 to merge once the build is green.

**Pull request merged**